xbmc/filesystem/CurlFile.cpp

```diff
@@ -925,6 +940,16 @@ void CCurlFile::Reset()
   m_state->m_cancelled = false;
 }
 
+void CCurlFile::SetProxy(ProxyType type, const std::string &host,
```

Comment:
Well I'll do what you're asking, if you're sure about that.

In my mind having to do string parsing over and over throughout the code-base is rather ugly, when it could just be parsed once. Again, I had in mind that I was setting up for broader proxy support in the future.

But if you're sure you want it to be string based, I can do that.